From 495d31773722a3ce2eb9c197eea9bbb1af619a61 Mon Sep 17 00:00:00 2001
From: Ye Li <ye.li@nxp.com>
Date: Wed, 13 Dec 2017 00:20:15 -0600
Subject: [PATCH] MLK-17200-2 mx7ulp: Remove SNVS LP settings for B0

Since i.MX7ULP B0 moves the SNVS LP into M4 domain, A core can't access
it. So check the CPU rev and not apply the settings for B0.

Signed-off-by: Ye Li <ye.li@nxp.com>
Reviewed-by: Peng Fan <peng.fan@nxp.com>
---
 arch/arm/cpu/armv7/mx7ulp/soc.c             | 10 ++++++----
 arch/arm/include/asm/arch-mx7ulp/imx-regs.h |  3 +++
 2 files changed, 9 insertions(+), 4 deletions(-)

diff --git a/arch/arm/cpu/armv7/mx7ulp/soc.c b/arch/arm/cpu/armv7/mx7ulp/soc.c
index 587835e..715a8ee 100644
--- a/arch/arm/cpu/armv7/mx7ulp/soc.c
+++ b/arch/arm/cpu/armv7/mx7ulp/soc.c
@@ -152,13 +152,15 @@ void s_init(void)
 	/* clock configuration. */
 	clock_init();
 
-	/* enable dumb pmic */
-	writel((readl(SNVS_LP_LPCR) | 0x20), SNVS_LP_LPCR);
+	if (soc_rev() < CHIP_REV_2_0) {
+		/* enable dumb pmic */
+		writel((readl(SNVS_LP_LPCR) | SNVS_LPCR_DPEN), SNVS_LP_LPCR);
 
 #if defined(CONFIG_ANDROID_SUPPORT)
-        /* Enable RTC */
-        writel(0x21, 0x40230038);
+		/* Enable RTC */
+		writel((readl(SNVS_LP_LPCR) | SNVS_LPCR_SRTC_ENV), SNVS_LP_LPCR);
 #endif
+	}
 	return;
 }
 
diff --git a/arch/arm/include/asm/arch-mx7ulp/imx-regs.h b/arch/arm/include/asm/arch-mx7ulp/imx-regs.h
index ea71ffb..0305eff1 100644
--- a/arch/arm/include/asm/arch-mx7ulp/imx-regs.h
+++ b/arch/arm/include/asm/arch-mx7ulp/imx-regs.h
@@ -949,6 +949,9 @@
 #define MMDC_MPWRDQBY3DL_WR_DQ25_DEL_MASK	((0x3f  << MMDC_MPWRDQBY3DL_WR_DQ25_DEL))
 #define MMDC_MPWRDQBY3DL_WR_DQ24_DEL_MASK	((0x3f  << MMDC_MPWRDQBY3DL_WR_DQ24_DEL))
 
+#define SNVS_LPCR_DPEN				(0x20)
+#define SNVS_LPCR_SRTC_ENV			(0x1)
+
 #include <asm/imx-common/regs-lcdif.h>
 
 #if !(defined(__KERNEL_STRICT_NAMES) || defined(__ASSEMBLY__))
-- 
2.9.2

